### ISSUE TITLE
2.0: release what you have now?

### ISSUE THREAD

**Issue by dcnieho** [pyimgui/pyimgui#299]
I have been using the 2.0 branch with much enjoyment and made a nice GUI for my app. I plan to distribute this both as an executable with cx_Freeze (works well) and on PyPI. Its the latter where i run into an issue. Upon trying to upload, i am greeted with:
```
HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/        
Invalid value for requires_dist. Error: Can't have direct dependency:  
'imgui @ git+https://github.com/pyimgui/pyimgui.git@dev/version-2.0'
```
The problem here being that since 2.0 hasn't been released, i have to refer to the github, not a pypi package version of pyimgui. And PyPI doesn't allow that, even if pip thinks its fine.

So i would like to ask: would you consider releasing 2.0 as it is now, even if it isn't complete? I mean, the old 1.x isn't complete either, and 2.0 is completely usable and has some nice new functionality thanks to the much newer imgui version that i'd love to use.

On a semi-related note: there are some functions i'd like to use that aren't wrapped yet. I might look into doing this myself. Is the code generated with some tool that is not part of the repo, or is it a lot of boilerplate i should manually repeat?

**Comment by KinoxKlark** [comment]
Yes, I may consider releasing it. The initial plane was to fix some build pipeline issues and to compile the cpp version as a .dll before releasing 2.0 since it could potentially break stuff if introduced in a minor release. But time flies and I haven't had time to work on it and probably won't have time in a near future. Maybe it is indeed more reasonable to release the 2.0 as it is for everyone to enjoy it. The last thing I want is to block you in your project. Moreover, it could bring back some momentum to the project. (@learn-more what is your opinion on the subject?)

I must manage some PR and issue in coming days. I will also look for putting something together for this release. 

For the unwrapped functions: We don't have tools to generate code. It is all done by hand. Every mappable function of DearImGui is listed in `cimgui.pxd` and is marked with `✓` or `✗` to indicate if it has been implemented or not. This system allows to compute the ratio of mapped function vs. unmapped function. 

For now, version 2.0 is based on DearImGui v1.82. If the functions you want to implement are part of this release you should find it in the `cimgui.pxd` file marked as `✗`. You can then implement it in the `core.pyx` and change the marking to `✓`. If it is not part of 1.82 but from a more recent release it will need a bit more work. We would have to update the cpp lib and update the `cimgui.pxd` (adding new functions, adding new flags, removing old one, marking them as depreciated, etc.) which is a lot of manual work. To do so I use to proceed step by step and to only upgrade one release at the time instead of jumping several releases. And the work must be done in parallel for the `docking` branch. We have a list of corresponding commits here which is nearly complete #259 (I have to add info for v1.88). If we release 2.0 as it is, I may consider updating to 1.88 for the next `dev` branch. What functions would you want to map?

**Comment by learn-more** [comment]
@KinoxKlark releasing 2.0 as-is seems like a reasonable step forward.
The big jump in Dear ImGui version is nicely communicated with the major release bump.
If there is a working solution ready, this can be in the next release,
which can then be 2.x or even a 3.x

**Comment by mafried** [comment]
Hi, any news on that? My hope is that the new release fixes some high DPI issues on MacOS so I can't wait 👍

**Comment by jamesdolezal** [comment]
I know this issue is a bit old, but I haven't seen another place for discussing the current status and future of the project.

Pyimgui has been an incredibly helpful library for our work. We built a medical imaging application for open-source, deep-learning powered microscopes using this framework, called [Slideflow Studio](http://slideflow.dev/studio/). I tried to get the more popular DearPyGui framework to work for us, given it's greater popularity, but it didn't offer the same flexibility as pyimgui. Pyimgui's immediate mode, greater responsiveness, and ability to provide access to lower-level OpenGL functions directly has made it a much more useable package for larger projects. 

I wanted to touch base about the future of the project, as it appears that pyimgui is no longer under active development. Can we muster a group of people willing to assemble the `dev` branch into a stable 2.0 release? I do believe that pyimgui offers significant advantages compared to DearPyGui and it's definitely worth continuing development, if we can find the manpower to do so.

**Comment by KinoxKlark** [comment]
Thank you for reaching out. It is fantastic to see what is being done using pyimgui. Thanks for sharing!

I am deeply sorry for the lack of an update on that. I am refusing to see the project as being dead, but the truth is that I am struggling to find free time to work on it, and my motivation is slowly drifting away. I will try my best to book some time for releasing 2.0 as it is. As I remember, it is basically ready, and I have to gather everything together, proceed with last-minute checks, and release it to the world. There are several works in progress, like the dll functionality and others, that I have to look at. But I believe it could be released as it is such that people can use it by default.

My regrets are that:
1. It has been so long without releasing that this 2.0 update that was supposed to update the dependency DearImGui to a more recent release is now already based on an old release. 2.0 is targeting DearImGui 1.82, and 1.89.5 just released.
2. Each submodule of pyimgui (`core` and `internal`) each have one full copy of the cpp library due to the hacky way it has been implemented. It does not seem solid enough to be released, and fixing this was the main point of the dll feature. But it is a whole project in itself, and I clearly won't be able to do it if I want to release 2.0 shortly. 

If a group of motivated contributors came together, I would be willing to give some access to the github organization for the project to continue its life.

**Comment by KinoxKlark** [comment]
Things are finally moving. I just merged `dev/version-2.0` branch into master. I had to make some last-minute changes this weekend to ensure compatibility with python 3.10 and 3.11. I am now waiting for the build and then I will try to write a nice release note and release it. Thanks everyone for your precious help ;)

**Comment by KinoxKlark** [comment]
I finally released it! (See: https://github.com/pyimgui/pyimgui/releases/tag/2.0.0)

They will probably be some bug to fix in coming days. I will try my best to take some time for that, but it will not be easy. If some generous and motivated contributors are willing to help it could be wonderful.

Thanks to everyone who already helped with the release!

(I'll close this issue since it has been "solved")

**Comment by jamesdolezal** [comment]
Congratulations! This was a huge effort an excellent work. I can definitely contribute to testing and bug fixing.

Thank you again for all your work on this package.
